src/v/utils/retry_chain_node.h

```diff
@@ -200,8 +200,9 @@ class retry_chain_node {
     using milliseconds_uint16_t
       = std::chrono::duration<uint16_t, std::chrono::milliseconds::period>;
 
-    /// Create a head of the chain without backoff
-    retry_chain_node();
+    // No default constructor: we always need an abort source.
+    retry_chain_node() = delete;
```

Comment:
nit: type of `_parent` can be changed now since we will never use `std::monostate` variant.